src/v/pandaproxy/handlers.cc

```diff
@@ -108,12 +108,11 @@ get_topics_records(server::request_t rq, server::reply_t rp) {
     rq.req.reset();
     return rq.ctx.client
       .fetch_partition(std::move(tp), offset, max_bytes, timeout)
-      .then([fmt,
-             rp = std::move(rp)](kafka::fetch_response::partition res) mutable {
+      .then([fmt, rp = std::move(rp)](kafka::fetch_response res) mutable {
           rapidjson::StringBuffer str_buf;
           rapidjson::Writer<rapidjson::StringBuffer> w(str_buf);
 
-          ppj::rjson_serialize_fmt(fmt)(w, std::move(res));
+          ppj::rjson_serialize_fmt(fmt)(w, std::move(res.partitions[0]));
```

Comment:
why does the json handler here only take the first partition? (in fairness it looks like this is preserving the behavior from before, just curious).

Reply:
I was trying to minimise the diff/preserve behaviour, but I'll fix this.